src/main/java/core/basesyntax/strategy/impl/TransactionStrategyImpl.java

```diff
@@ -0,0 +1,29 @@
+package core.basesyntax.strategy.impl;
+
+import core.basesyntax.db.FruitStorage;
+import core.basesyntax.handlers.TransactionHandler;
+import core.basesyntax.handlers.impl.RemoveTransactionHandler;
+import core.basesyntax.handlers.impl.SaveTransactionHandler;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.strategy.TransactionStrategy;
+import java.util.HashMap;
+import java.util.Map;
+
+public class TransactionStrategyImpl implements TransactionStrategy {
+    private final Map<FruitTransaction.Operation, TransactionHandler> strategyMap;
+
+    public TransactionStrategyImpl(FruitStorage storage) {
+        this.strategyMap = new HashMap<>();
+        TransactionHandler saveHandler = new SaveTransactionHandler(storage);
+        TransactionHandler removeHandler = new RemoveTransactionHandler(storage);
+        strategyMap.put(FruitTransaction.Operation.BALANCE, saveHandler);
+        strategyMap.put(FruitTransaction.Operation.RETURN, saveHandler);
+        strategyMap.put(FruitTransaction.Operation.PURCHASE, removeHandler);
+        strategyMap.put(FruitTransaction.Operation.SUPPLY, saveHandler);
+    }
```

Comment:
better move this in Main main method and pass map in constructor of TransactionStrategyImp